### PR TITLE
Port to Python 3

### LINKED ISSUE #787: Port to Python 3
Port Sugar to Python 3 and the Sugar Toolkit for GTK+ 3, providing a Python 3 implementation of this package with no Python 2 dependencies.

Requires most of https://github.com/sugarlabs/sugar-toolkit-gtk3/issues/382, most of https://github.com/sugarlabs/sugar-datastore/issues/7 and some of https://github.com/sugarlabs/hello-world-fork/issues/2 to be completed.

Progress goals:

- [x] port code in `src/jarabe`,
- [x] port code in `extensions`,
- [x] test display of available activities, which may be Python 2 or Python 3,
- [x] test starting the [Hello World activity](https://github.com/sugarlabs/hello-world-fork), a Python 3 GTK+ 3 activity,
- [x] test starting a Python 2 GTK+ 3 activity,
- [x] test starting a Python 2 GTK+ 2 activity,
- [x] port make files, and test that "make install" installs Python 3 binding.
- [x] update in-tree documentation,
- [x] fix newly introduced bugs,
- [x] <s>make local Fedora and Debian packages for testing by others.</s>

_this is an issue for tracking a major project which is larger than one person's contributions, so developers should make pull requests to satisfy any of these items, and not ask to own the issue._

### PR DESCRIPTION
**Pull request by pro-panda**
Fixes #787

### CODE REVIEW

**Comment by quozl** [comment]
Thanks.  Reviewed to 8166eb0.  Not tested.  Lost of symbolic constants for `os.statvfs` is unfortunate; is there any alternative?

**Comment by pro-panda** [comment]
> is there any alternative?

None that I could see

**Comment by manddy** [comment]
@pro-panda  i am not getting why there is merge conflicts in src/jarabe/controlpanel/cmd.py?

**Comment by quozl** [comment]
Did `git checkout python3`, `git rebase -i master`, resolved conflicts, and `git push --force`.

**Comment by manddy** [comment]
@quozl  Is this issue complete or  there is anything else to still work on?

File: src/jarabe/desktop/viewtoolbar.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def _entry_activated_cb(self, entry):
         new_query = entry.props.text
         if self._query != new_query:
             self._query = new_query
+            if isinstance(self._query, bytes):
```
Comment (manddy) [inline src/jarabe/desktop/viewtoolbar.py]: @pro-panda   why did you add this line in the code,because it not  not related to this issue of porting to a different version @quozl Your  suggestions will also be helpful

Comment (quozl) [reply src/jarabe/desktop/viewtoolbar.py]: @manddy, thanks for asking.  The change seems common to me.  @pro-panda didn't explain the change in the commit message.  I'm not sure why the change is needed though.  I've no suggestions.

Comment (pro-panda) [reply src/jarabe/desktop/viewtoolbar.py]: Thanks for asking @manddy.
Most of such changes were introduced since https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/383 supported both Python 2 and 3, which implied that you could expect `bytes` or `unicode`, depending on the version an interacting sugar activity is written in.

From what I recall, this instance was also implemented for a similar reason.

Comment (quozl) [reply src/jarabe/desktop/viewtoolbar.py]: Oh, of course, thanks.

**Comment by Aniket21mathur** [comment]
em.py is removed from master, as well as _popwindow_ commit is merged. Rebased and resolved conflicts with master. #844 .
